Tasks/PublishTestResults/publishtestresults.ts

```diff
@@ -18,23 +19,42 @@ tl.debug('config: ' + config);
 tl.debug('testRunTitle: ' + testRunTitle);
 tl.debug('publishRunAttachments: ' + publishRunAttachments);
 
-//check for pattern in testResultsFiles
-if(testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
-  tl.debug('Pattern found in testResultsFiles parameter');
-  var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
-  var allFiles = tl.find(buildFolder);
-  var matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, { matchBase: true });
-}
-else {
-  tl.debug('No pattern found in testResultsFiles parameter');
-  var matchingTestResultsFiles = [testResultsFiles];
-}
+var matchingTestResultsFiles: string[] = getMatchingFiles(tl.getVariable('System.DefaultWorkingDirectory'), testResultsFiles, 'testResultsFiles');
 
-if(!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
-  tl.warning('No test result files matching ' + testResultsFiles + ' were found.');  
+if (!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
+  tl.warning('No test result files matching ' + testResultsFiles + ' were found.');
   tl.exit(0);
 }
-else{
+else {
   var tp = new tl.TestPublisher(testRunner);
   tp.publish(matchingTestResultsFiles, mergeResults, platform, config, testRunTitle, publishRunAttachments);
+}
+
+function getMatchingFiles(baseDirectory: string, inputPattern: string, inputPatternName: string): string[] {
```

Comment:
This function is trying to get base path from pattern which could be outside basedirectory. There is similar requirement in vstest task for this and I had written almost same logic. Refer: [https://github.com/Microsoft/vsts-tasks/pull/2287/files](https://github.com/Microsoft/vsts-tasks/pull/2287/files) for vstest task. Can we make this a generic method that is to get absolute base path from the file pattern, then it can be used by multiple tasks